MAUI/DataGrid/sorting.md

```diff
@@ -0,0 +1,252 @@
+platform: MAUI
+control: SfDataGrid
+documentation: UG
+---
+
+# Sorting in MAUI DataGrid (SfDataGrid)
+
+The [SfDataGrid](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html) provides the built-in support to sort one or more columns by using the [SfDataGrid.SortingMode](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_SortingMode) property. When sorting is applied, the datagrid automatically rearranges the data to match with the current sort criteria. You can sort the data simply by tapping the column header. Once sorting is applied, the datagrid shows a sort icon in the respective column header indicating the sort direction.
+
+## Programmatic sorting
+
+You can sort the data programmatically by adding or removing the [SortColumnDescription](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SortColumnDescription.html) in [SfDataGrid.SortColumnDescriptions](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_SortColumnDescriptions) property.
+
+The `SortColumnDescription` object holds the following two properties,
+
+* **ColumnName**: Name of the sorted column.
+* **SortDirection**: An object of type `ListSortDirection` defines the sorting direction.
+
+{% tabs %}
+{% highlight xaml %}
+<syncfusion:SfDataGrid x:Name="sfDataGrid" ItemsSource="{Binding OrderInfoCollection}" >
+
+    <syncfusion:SfDataGrid.SortColumnDescriptions>
+        <syncfusion:SortColumnDescription ColumnName="OrderID" SortDirection="Ascending" />
+    </syncfusion:SfDataGrid.SortColumnDescriptions>
+</syncfusion:SfDataGrid>
+{% endhighlight %}
+
+{% highlight c# %}
+SortColumnDescription sortColumnDescription = new SortColumnDescription() { ColumnName = "OrderID", SortDirection = ListSortDirection.Ascending };
+this.sfDataGrid.SortColumnDescriptions.Add(sortColumnDescription);
+{% endhighlight %}
+{% endtabs %}
+
+![Programmatical sorting .NET MAUI DataGrid](Images\sorting\maui-datagrid-programmatic-sorting.png)
+
+## Different modes of sorting
+
+`SfDataGrid` sorts the data against one or more columns based on the `SfDataGrid.SortingMode` property. The `SortingMode` contains the following properties,
+
+* **Single** - It allows to sort only one column at a time.
+* **Multiple** - It allows to sort more than one column at a time.
+* **None** - None of the columns can be sorted.
+
+To apply sorting for multiple columns, tap the desired column headers after setting the `SfDataGrid.SortingMode` property to [Multiple](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridSelectionMode.html#Syncfusion_Maui_DataGrid_DataGridSelectionMode_Multiple).
+
+{% tabs %}
+{% highlight xaml %}
+<syncfusion:SfDataGrid x:Name="sfDataGrid" SortingMode="Multiple" ItemsSource="{Binding OrderInfoCollection}" >
+
+    <syncfusion:SfDataGrid.SortColumnDescriptions>

```

Comment:
@ashok-kuvaraja  - This programmatic sorting code is not required.

Reply:
Modified